src/community/params-extractor/src/main/java/org/geoserver/params/extractor/Filter.java

```diff
@@ -7,21 +7,35 @@
 import java.io.IOException;
 import java.util.List;
 import java.util.logging.Logger;
-import javax.servlet.*;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
 import javax.servlet.http.HttpServletRequest;
 import org.geoserver.config.GeoServerDataDirectory;
 import org.geoserver.filters.GeoServerFilter;
 import org.geoserver.platform.ExtensionPriority;
+import org.geoserver.platform.GeoServerExtensions;
 import org.geoserver.platform.resource.Resource;
 import org.geotools.util.logging.Logging;
 
 public final class Filter implements GeoServerFilter, ExtensionPriority {
 
     private static final Logger LOGGER = Logging.getLogger(Filter.class);
 
+    static boolean USE_AS_SERVLET_FILTER = false;
+
+    private boolean servletInstance = false;
```

Comment:
Two flags... so the static one is used to know that there is an instance used as a servlet filter, and the other one to know which one is the servlet filter, to disable the other? I'd add comments to clarify it, was not obvious to me at a first read.